OpenRA.Mods.Common/MapGenerator/TilingPath.cs

```diff
@@ -1191,5 +1238,42 @@ public static CPos[][] RetainDisjointPaths(IEnumerable<CPos[]> inputs)
 
 			return outputs.ToArray();
 		}
+
+		/// <summary>Nullify the path's points if they aren't suitable for tiling.</summary>
+		public TilingPath RetainIfValid()
+		{
+			if (!ValidatePathPoints(Points))
+				Points = null;
+
+			return this;
+		}
+
+		public static bool ValidatePathPoints(CPos[] points)
+		{
+			if (points == null || points.Length == 0)
+				return false;
+
+			var isLoop = points[0] == points[^1];
+
+			if (points.Length < (isLoop ? 3 : 2))
+				return false;
+
+			// Duplicate points check
+			if (points.ToHashSet().Count != points.Length - (isLoop ? 1 : 0))
```

Comment:
```suggestion
			if (points.Distinct().Count() != points.Length - (isLoop ? 1 : 0))
```

This is the standard way of doing it. It also uses a set, but the higher-level optimization might pick up on intent better